homeassistant/components/uvc/camera.py

```diff
@@ -97,14 +103,31 @@ def name(self):
     @property
     def is_recording(self):
         """Return true if the camera is recording."""
-        caminfo = self._nvr.get_camera(self._uuid)
-        return caminfo['recordingSettings']['fullTimeRecordEnabled']
+        return self._caminfo['recordingSettings']['fullTimeRecordEnabled']
 
     @property
     def motion_detection_enabled(self):
         """Camera Motion Detection Status."""
-        caminfo = self._nvr.get_camera(self._uuid)
-        return caminfo['recordingSettings']['motionRecordEnabled']
+        return self._caminfo['recordingSettings']['motionRecordEnabled']
+
+    @property
+    def supported_features(self):
+        """Return supported features."""
+        for channel in self._caminfo['channels']:
+            if channel['isRtspEnabled']:
+                return SUPPORT_STREAM
+        return 0
+
+    @property
+    def stream_source(self):
+        """Return the stream source."""
+        for channel in self._caminfo['channels']:
+            if channel['isRtspEnabled']:
+                stream = channel['rtspUris'][0].split(':')[2]
+                return 'rtsp://{}:{}'.format(
+                    self._caminfo['controllerHostAddress'],
+                    stream)
+            return None
```

Comment:
You return `None` inside the for loop, so only handle 1 channel?

Reply:
moved to get_camera_info.